backend/src/main/java/ca/bc/gov/app/service/client/ClientService.java

```diff
@@ -141,6 +142,76 @@ public Mono<ClientDetailsDto> getClientDetails(
             "Unable to process request. This sole proprietor is not owned by a person"
         )));
   }
+  
+  public Mono<ForestClientDetailsDto> getClientDetailsByClientNumber(
+      String clientNumber,
+      List<String> groups
+  ) {
+      log.info("Loading details for {} for user role {}", clientNumber, groups.toString());
+
+      return legacyService
+          .searchByClientNumber(clientNumber, groups)
+          .flatMap(forestClientDetailsDto -> {
+              String corpRegnNmbr = forestClientDetailsDto.corpRegnNmbr();
+
+              if (corpRegnNmbr == null || corpRegnNmbr.isEmpty()) {
+                  log.info("Corporation registration number not provided. Returning legacy details.");
+                  return Mono.just(forestClientDetailsDto);
+              }
+
+              log.info("Retrieved corporation registration number: {}", corpRegnNmbr);
+
+              Mono<BcRegistryDocumentDto> documentMono = bcRegistryService
+                  .requestDocumentData(corpRegnNmbr)
+                  .next();
+              return populateGoodStandingInd(forestClientDetailsDto, documentMono);
+          });
+  }
+
+  private Mono<ForestClientDetailsDto> populateGoodStandingInd(
+      ForestClientDetailsDto forestClientDetailsDto,
+      Mono<BcRegistryDocumentDto> documentMono
+  ) {
+      return documentMono
```

Comment:
As this specific method content does not consume another mono other than the one provided as a parameter, it can be changed to not use mono at all and the above method call can be a `.map` as is being used here instead of the proposed `.flatMap`